blueflood-core/src/main/java/com/rackspacecloud/blueflood/cache/ConfigTtlProvider.java

```diff
@@ -41,98 +43,105 @@ public static ConfigTtlProvider getInstance() {
     }
 
     private ConfigTtlProvider() {
-        final Configuration config = Configuration.getInstance();
+        this(Configuration.getInstance());
+    }
 
+    @VisibleForTesting
+    ConfigTtlProvider(Configuration config) {
         // String rollups
-        stringTTL = new TimeValue(config.getIntegerProperty(TtlConfig.STRING_METRICS_TTL), TimeUnit.DAYS);
+        TimeValue stringTTL = null;
+        try {
+            int value = config.getIntegerProperty(TtlConfig.STRING_METRICS_TTL);
+            if (value > 0) {
+                stringTTL = new TimeValue(config.getIntegerProperty(TtlConfig.STRING_METRICS_TTL), TimeUnit.DAYS);
+            }
+        } catch (NumberFormatException ex) {
+            log.warn("No valid String TTL in config.");
```

Comment:
I'd include the exception as well.  E.g., http://www.slf4j.org/api/org/slf4j/Logger.html#warn(java.lang.String, java.lang.Throwable)